clients/web/src/rest.js

```diff
@@ -171,6 +172,12 @@ let restRequest = function (opts) {
         throw new Error('restRequest requires a "url" argument');
     }
     args.url = `${getApiRoot()}${args.url.substring(0, 1) === '/' ? '' : '/'}${args.url}`;
+    // jQuery's ajax method prefers `method` over `type`.  Since we specify a
+    // default `method`, if the caller supplies a `type`, use it instead.
+    if (args.type) {
```

Comment:
The original intent of #2273 was to bring the behavior of `restRequest` closer to `$.ajax`, and make it documented and explicit that `restRequest` is a thin wrapper around `$.ajax` (previously, some code was abusing this behavior, even though it was never documented). For libraries like `jQuery` that are well documented, broadly understood, and "leaked" to effectively be part of Girder's API, I think it's more sensible to delegate to their defaults whenever possible, which both decreases the LOC (and associated maintenance burden) in Girder and makes it easier for new developers to learn Girder, as they're more likely to understand `jQuery` or be able to effectively search StackOverflow, etc. for help.

@manthey What do you think of this?